### PR TITLE
fix translation request handling (WP-609)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**
I've previously introduced a regression in handling, while fixing 2 levels deep cloning issues. Fixed, tests added

### CODE REVIEW

File: inc/Smartling/ContentTypes/AutoDiscover/PostTypes.php [hunk 1]
```diff
@@ -17,7 +19,7 @@ public function hookHandler($postType): void
             return;
         }
 
-        add_action('smartling_register_custom_type', static function (array $definition) use ($postType) {
+        add_action(ExportedAPI::FILTER_SMARTLING_REGISTER_CUSTOM_POST_TYPE, static function (array $definition) use ($postType) {
```
Comment (vsolovei-smartling) [inline inc/Smartling/ContentTypes/AutoDiscover/PostTypes.php:22]: Not related to the WP-609 issue, but I don't really wan't to create a PR just for this single change

File: inc/Smartling/Models/CloneRequest.php [hunk 2]
```diff
@@ -45,4 +43,19 @@ public function getTargetBlogIds(): array
     {
         return $this->targetBlogIds;
     }
+
+    public static function fromArray(array $array): self
+    {
+        return new self(self::getSourceId($array), $array['source']['contentType'], $array['relations'] ?? [], explode(',', $array['targetBlogIds']));
```
Comment (PavelLoparev) [inline inc/Smartling/Models/CloneRequest.php:49]: What if arry doesn't contain `['source']['contentType']`?

File: inc/Smartling/Models/TranslationRequest.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+class TranslationRequest extends CloneRequest
+{
+    private JobInformation $jobInformation;
+    private array $ids;
+
+    public function __construct(int $contentId, string $contentType, array $relations, array $targetBlogIds, JobInformation $jobInformation, array $ids)
+    {
+        parent::__construct($contentId, $contentType, $relations, $targetBlogIds);
+        $this->jobInformation = $jobInformation;
+        $this->ids = self::toIntegerArray($ids);
```
Comment (PavelLoparev) [inline inc/Smartling/Models/TranslationRequest.php:16]: Why do we need own `toIntegerArray` defined in this class? Why not dyrectly use `ArrayHelper::toArrayOfIntegers()` here?

Comment (vsolovei-smartling) [reply inc/Smartling/Models/TranslationRequest.php:16]: To avoid writing the same error message twice

**Review by PavelLoparev — APPROVED** [review]

**Pull request merged**